packages/backend/src/open_payments/payment/outgoing/service.ts

```diff
@@ -118,6 +127,35 @@ export function isCreateFromIncomingPayment(
   return 'incomingPayment' in options && 'debitAmount' in options
 }
 
+async function cancelOutgoingPayment(
+  deps: ServiceDependencies,
+  options: CancelOutgoingPaymentOptions
+): Promise<OutgoingPayment | OutgoingPaymentError> {
+  const { id } = options
+
+  return deps.knex.transaction(async (trx) => {
+    let payment = await OutgoingPayment.query(trx).findById(id).forUpdate()
+
+    if (!payment) return OutgoingPaymentError.UnknownPayment
+    if (payment.state !== OutgoingPaymentState.Funding) {
+      return OutgoingPaymentError.WrongState
+    }
+
+    payment = await payment
+      .$query(trx)
+      .patchAndFetch({
+        state: OutgoingPaymentState.Cancelled,
+        metadata: {
+          ...payment.metadata,
+          cancellationReason: options.reason
+        }
+      })
```

Comment:
```suggestion
        metadata: {
          ...payment.metadata,
          ...(options.reason ? { cancellationReason: options.reason } : {})
        }
      })
```
so we don't set `cancellationReason: undefined`

Reply:
> if the field is undefined, then it will not be included in the metadata when doing serialization to database right

This one I'm not sure about since `metadata` is stored as JSON in the DB, and maybe it doesn't mind saving values as undefined